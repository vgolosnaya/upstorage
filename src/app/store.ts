import { configureStore } from '@reduxjs/toolkit';
import filesReducer from './file/files.reducer';

export const store = configureStore({
    reducer: {
        files: filesReducer,
    },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
