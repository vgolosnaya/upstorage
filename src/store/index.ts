import { combineReducers, configureStore } from '@reduxjs/toolkit';
import filesReducer from 'store/files/files.reducer';
import { IFile } from 'services/FileReader.service';
import filesMiddleware from 'store/files/files.middleware';
import StorageService from 'services/Storage.service';
import errorsReducer, { ErrorState } from 'store/errors/errors.reducer';

const storageService = new StorageService<IFile>('files');

export const store = configureStore({
    reducer: combineReducers({
        files: filesReducer,
        errors: errorsReducer,
    }),
    middleware: [ filesMiddleware(storageService) ]
});

export type AppDispatch = typeof store.dispatch;
export type RootState = {
    files: IFile[],
    errors: ErrorState[],
};
