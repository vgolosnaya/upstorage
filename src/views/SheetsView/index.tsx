import React from 'react';
import {
    Paper, Table, TableBody, TableCell, TableContainer, TableHead, TableRow, Typography
} from '@mui/material';
import LABELS from '../../labels';
import filesSelectors from '../../app/files/files.selectors';
import { useAppSelector } from '../../app/hooks';

const SheetsView = (): JSX.Element => {
    const csvFiles = useAppSelector(filesSelectors.getCsv);
    
    return (
        <div className="container">
            <TableContainer component={Paper}>
                <Table sx={{ minWidth: 650 }} aria-label="simple table">
                    <TableHead>
                        <TableRow>
                            <TableCell>FileName</TableCell>
                            <TableCell align="right">Total</TableCell>
                            <TableCell align="right">Datetime</TableCell>
                        </TableRow>
                    </TableHead>
                    <TableBody>
                        {csvFiles.map((file) => (
                            <TableRow
                                key={file.id}
                                sx={{ '&:last-child td, &:last-child th': { border: 0 } }}
                            >
                                <TableCell component="th" scope="row">
                                    {file.fileName}
                                </TableCell>
                                <TableCell align="right">{file.total}</TableCell>
                                <TableCell align="right">{file.relativeDateTime}</TableCell>
                            </TableRow>
                        ))}
                    </TableBody>
                </Table>
            </TableContainer>
        </div>
    );
};

export default SheetsView;
