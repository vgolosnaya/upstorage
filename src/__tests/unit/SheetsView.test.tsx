import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { store } from 'store';
import React from 'react';
import SheetsView from 'views/SheetsView';

describe('[SNAPSHOT] Sheets page:', ()=>{
    test('Should match snapshot', () => {
        const component = render(
            <Provider store={store}>
                <SheetsView/>
            </Provider>
        );
        
        expect(component.baseElement).toMatchSnapshot();
    });
});