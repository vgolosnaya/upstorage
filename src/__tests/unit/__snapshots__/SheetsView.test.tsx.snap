// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[SNAPSHOT] Sheets page: Should match snapshot 1`] = `
<body>
  <div>
    <div
      class="container"
    >
      <span
        class="MuiTypography-root MuiTypography-overline css-b3fch5-MuiTypography-root"
      >
        You don't have any file loaded
      </span>
    </div>
  </div>
</body>
`;
